File: IoC/app/src/main/java/com/lee/code/ioc/base/BaseActivity.java
package com.lee.code.ioc.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.lee.library.ioc.InjectManager;

/**
 * @author jv.lee
 */
public abstract class BaseActivity extends AppCompatActivity {


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        //帮助所有子类进行：布局/控件/事件的注入
        InjectManager.inject(this);
        bindData(savedInstanceState);
    }

    public abstract void bindData(Bundle savedInstanceState);
}

File: KotlinLibraryX/library/src/main/java/com/lee/library/base/BaseActivity.kt
package com.lee.library.base

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import com.lee.library.tools.SystemBarTools.navigationPadding
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch

/**
 * Activity通用基类
 * @author jv.lee
 * @date 2021/6/15
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        window.navigationPadding()
        super.onCreate(savedInstanceState)

        initSavedState(intent, savedInstanceState)

        bindView()

        bindData()
    }

    open fun initSavedState(intent: Intent, savedInstanceState: Bundle?) {}

    protected abstract fun bindView()

    protected abstract fun bindData()

    fun launchOnLifecycle(
        state: Lifecycle.State = Lifecycle.State.CREATED,
        block: suspend CoroutineScope.() -> Unit
    ) {
        lifecycleScope.launch {
            repeatOnLifecycle(state, block)
        }
    }
}